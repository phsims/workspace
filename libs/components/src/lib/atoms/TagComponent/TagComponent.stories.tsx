import type { Meta, StoryObj } from '@storybook/react';
import { TagComponent } from './TagComponent';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof TagComponent> = {
  component: TagComponent,
  title: 'TagComponent',
};
export default meta;
type Story = StoryObj<typeof TagComponent>;

export const Primary = {
  args: {},
};

export const Heading: Story = {
  args: {},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to TagComponent!/gi)).toBeInTheDocument();
  },
};
