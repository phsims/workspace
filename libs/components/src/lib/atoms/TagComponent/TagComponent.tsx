import {
  Box,
  Typography,
  Chip,
  Stack,
} from '@mui/material';

export interface TagComponentProps {
  title: string;
  tags: string[];
  varient?: 'outlined' | 'filled';
  color?: 'default' | 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning';
}

export function TagComponent({
  title,
  tags,
  color = 'primary',
  varient = 'filled',
}: TagComponentProps) {
  return (
    <Box >
      <Typography variant="h6" mb='0.5rem'>{title}</Typography>
      <Stack spacing={1} alignItems="center" direction="row" flexWrap='wrap' useFlexGap>
        {tags.map((tag) => (
          <Chip label={tag} color={color} variant={varient}/>
        ))}
      </Stack>
    </Box>
  );
}

export default TagComponent;
