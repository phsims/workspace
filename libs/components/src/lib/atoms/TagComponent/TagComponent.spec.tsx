import { render } from '@testing-library/react';

import TagComponent from './TagComponent';

describe('TagComponent', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<TagComponent />);
    expect(baseElement).toBeTruthy();
  });
});
