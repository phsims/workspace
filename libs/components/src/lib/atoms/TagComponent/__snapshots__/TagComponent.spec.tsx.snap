// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagComponent should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1kx63bm-MuiTypography-root"
      >
        Welcome to TagComponent!
      </h6>
      <div
        class="MuiStack-root css-1dtqmwz-MuiStack-root"
      >
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1016kjc-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            tag1
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1016kjc-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            tag2
          </span>
        </div>
        <div
          class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-1016kjc-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            tag3
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
